### PR TITLE
fix: add get public object auth

### PR DESCRIPTION
**Pull request by joeylichang**
### Description

Add get public object auth logic.

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 

N/A

### CODE REVIEW

File: modular/gater/object_handler.go [hunk 1]
```diff
@@ -140,12 +140,24 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 		}
 		log.CtxDebugw(reqCtx.Context(), reqCtx.String())
 	}()
-
+	
 	reqCtx, err = NewRequestContext(r)
 	if err != nil {
-		return
+		// check the object's visibility type whether equal to public read
+		authorized, err = g.baseApp.Consensus().VerifyGetObjectPermission(reqCtx.Context(), "",
```
Comment (Fynnss) [inline modular/gater/object_handler.go:146]: What if the user has GetObject permission and the object is private?

Comment (joeylichang) [reply modular/gater/object_handler.go:146]: new request context return err == nil

File: modular/gater/object_handler.go [hunk 2]
```diff
@@ -140,12 +140,24 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
...
 	}()
-
 	reqCtx, err = NewRequestContext(r)
 	if err != nil {
-		return
+		// check the object's visibility type whether equal to public read
+		authorized, err = g.baseApp.Consensus().VerifyGetObjectPermission(reqCtx.Context(), "",
+			reqCtx.bucketName, reqCtx.objectName)
+		if err != nil {
+			log.CtxErrorw(reqCtx.Context(), "failed to verify authorize for getting public object", "error", err)
```
Comment (sysvm) [inline modular/gater/object_handler.go:149]: `failed to verify authorize` is fixed as `failed to verify authorization` is better. authorize is verb.

File: modular/gater/object_handler.go [hunk 3]
```diff
@@ -140,12 +140,24 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
...
+			err = ErrConsensus
+			return
+		}
+		if !authorized {
+			log.CtxErrorw(reqCtx.Context(), "no permission to operate, object is not public")
```
Comment (sysvm) [inline modular/gater/object_handler.go:154]: fix `"no permission to operate, object is not public"` as `"no permission to operate due to object is not public"`

File: modular/gater/object_handler.go (hunk 2 above)
Comment (joeylichang) [reply modular/gater/object_handler.go:149]: next pr

File: modular/gater/object_handler.go (hunk 3 above)
Comment (joeylichang) [reply modular/gater/object_handler.go:154]: next pr

File: modular/gater/object_handler.go (hunk 2 above)
Comment (joeylichang) [reply modular/gater/object_handler.go:149]: @will-2012 is doing log tidy in netx pr

File: modular/gater/object_handler.go (hunk 3 above)
Comment (joeylichang) [reply modular/gater/object_handler.go:154]: ditto

File: modular/gater/object_handler.go (hunk 1 above)
Comment (Fynnss) [reply modular/gater/object_handler.go:146]: Sorry, I didn't saw `VerifyPermission` in `NewRequestContext`.

**Pull request closed without merging**